src/main/java/no/sikt/nva/pubchannels/handler/CreateJournalHandler.java

```diff
@@ -50,13 +53,22 @@ public CreateJournalHandler(Environment environment, PublicationChannelClient pu
     @Override
     protected Void processInput(CreateJournalRequest input, RequestInfo requestInfo, Context context)
             throws ApiGatewayException {
+        var validInput = attempt(() -> validate(input))
+                .orElseThrow(failure -> new BadRequestException(failure.getException().getMessage()));
 
-        var journalPid = publicationChannelClient.createJournal(input.getName());
-        var createdUri = constructJournalIdUri(journalPid);
-        addAdditionalHeaders(() -> Map.of(HttpHeaders.LOCATION, createdUri.toString()));
+        var journalPid = publicationChannelClient.createJournal(validInput.getName());
+        addAdditionalHeaders(() -> Map.of(HttpHeaders.LOCATION, constructJournalIdUri(journalPid).toString()));
         return null;
     }
 
+    private CreateJournalRequest validate(CreateJournalRequest input) {
```

Comment:
As this is also a validation concern, why not move this to the Validator class too?

Reply:
I realize now that the validation from before is validating after being put in the request object for the client. 
I will fix this my extracting the validation of the input (or requestInfo) into the handler.